src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -566,7 +566,7 @@ recordDefinition![AST modifiers]
         ic:implementsClause
         rb:recordBodyDeclaration
         {#recordDefinition = #(#[RECORD_DEF, "RECORD_DEF"],
-                              modifiers, r, id, tp, ic, rc, rb);}
+                              modifiers, r, id, tp, rc, ic, rb);}
```

Comment:
@romani This is more than faulty indentation as issue implies. Our grammar was wrong and needs changing.